File: quackapp-frontend/src/main/angular/src/app/models/Comment.ts
import {User} from "./User";

export class Comment {
    id: number;
    text: string;
    date: Date;
    author: User;

    static fromObject(object: any) : Comment {
        const comment = new Comment();
        comment.id = object.id;
        comment.text = object.text;
        comment.date = new Date(parseInt(object.date));
        comment.author = User.fromObject(object.author);
        return comment;
    }
}

File: quackapp-backend/src/main/java/de/quackr/persistence/entities/Comment.java
package de.quackr.persistence.entities;

import javax.json.bind.annotation.JsonbDateFormat;
import javax.persistence.*;
import java.util.Date;

@Entity
public class Comment {

    @Id
    @GeneratedValue
    private long id;

    @ManyToOne
    private User author;

    @Column(nullable = false)
    private long qid;

    @Column(nullable = false)
    private String text;

    @Column(nullable = false)
    @Temporal(TemporalType.TIMESTAMP)
    @JsonbDateFormat(value = JsonbDateFormat.TIME_IN_MILLIS)
    private Date date;

	public long getId() {
		return id;
	}

	public void setId(long id) {
		this.id = id;
	}

	public User getAuthor() {
		return author;
	}

	public void setAuthor(User author) {
		this.author = author;
	}

	public long getQid() {
		return qid;
	}

	public void setQid(long qid) {
		this.qid = qid;
	}

	public String getText() {
		return text;
	}

	public void setText(String text) {
		this.text = text;
	}

    public Date getDate() {
        return date;
    }

    public void setDate(Date date) {
        this.date = date;
    }
}
